operator/builtin/input/aws/cloudwatch/cloudwatch.go

```diff
@@ -0,0 +1,322 @@
+	}
+
+	if c.Limit < 1 {
+		return nil, fmt.Errorf("invalid value '%d' for %s parameter 'limit'. Parameter 'limit' must be a value between 1 - 10000", c.Limit, operatorName)
+	}
+
+	if c.Limit > 10000 {
+		return nil, fmt.Errorf("invalid value '%d' for %s parameter 'limit'. Parameter 'limit' must be a value between 1 - 10000", c.Limit, operatorName)
+	}
+
+	if c.StartAt != "beginning" && c.StartAt != "end" {
+		return nil, fmt.Errorf("invalid value for parameter 'start_at'")
+	}
+
+	var startAtEnd bool
+	switch c.StartAt {
+	case "beginning":
+		startAtEnd = false
+	case "end":
+		startAtEnd = true
+	default:
+		return nil, fmt.Errorf("invalid value '%s' for %s parameter 'start_at'", c.StartAt, operatorName)
+	}
+
+	cloudwatchInput := &CloudwatchInput{
+		InputOperator:       inputOperator,
+		logGroupName:        c.LogGroupName,
+		logStreamNamePrefix: c.LogStreamNamePrefix,
+		region:              c.Region,
+		limit:               c.Limit,
+		profile:             c.Profile,
+		pollInterval:        c.PollInterval,
+		startAtEnd:          startAtEnd,
+		startTime:           0,
+		persist:             *helper.NewScopedDBPersister(buildContext.Database, c.ID()),
+	}
+	return []operator.Operator{cloudwatchInput}, nil
+}
+
+// CloudwatchInput is an operator that reads input from AWS Cloudwatch Logs.
+type CloudwatchInput struct {
+	helper.InputOperator
+	cancel       context.CancelFunc
+	pollInterval helper.Duration
+
+	logGroupName        string
+	logStreamNamePrefix string
+	region              string
+	limit               int64
+	profile             string
+	startAtEnd          bool
+	startTime           int64
+	persist             helper.ScopedBBoltPersister
+	wg                  sync.WaitGroup
+}
+
+// Start will start generating log entries.
+func (c *CloudwatchInput) Start() error {
+	ctx, cancel := context.WithCancel(context.Background())
+	c.cancel = cancel
+
+	if err := c.persist.Load(); err != nil {
+		return err
+	}
+
+	err := c.pollEvents(ctx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Stop will stop generating logs.
+func (c *CloudwatchInput) Stop() error {
+	c.cancel()
+	c.wg.Wait()
+	fmt.Printf("Closed all connections to Cloudwatch Logs")
+	return nil
+}
+
+// pollEvents gets events from AWS Cloudwatch Logs every poll interval.
+func (c *CloudwatchInput) pollEvents(ctx context.Context) error {
+	c.Info("Started polling AWS Cloudwatch using poll interval of ", c.pollInterval)
+	defer c.wg.Done()
+
+	region := aws.String(c.region)
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		Config: aws.Config{Region: region},
+
+		Profile: c.profile,
+	}))
+
+	svc := cloudwatchlogs.New(sess)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-time.After(c.pollInterval.Duration):
+			c.Debug("Getting events from AWS Cloudwatch Logs")
+			nextToken := ""
+			st := c.persist.Get(c.logGroupName)
+			c.startTime = BytesToInt64(st)
```

Comment:
I think we can wrap c.persist.Get() and Set() into helper methods that perform the correct conversion before returning values.

For example, theoretical implementaton:
```
type Persister {
  persist. helper.ScopedBBoltPersister
}

func (p *Persister) Read(key string) (int64, error) {
  x := p.persist.Get(key)

  if <some validation logic here, if it makes sense>

  return BytesToInt64(x), nil
}
```

This way, we can just do:
```
st, err := c.Persist.Read(c.logGroupName)
if err != nil {
  return err
}
c.startTime = st
```

Reply:
`BytesToInt64()` could be merged with Persister.Read(), and if you decide later to switch to json.Marshal / json.Unmarshal, you can easily update Read and Write without changing the logic that is relying on this implementation.